aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -100,56 +100,38 @@ static void _launch_kernel(int total_n_elems, func_t f) {
   C10_CUDA_KERNEL_LAUNCH_CHECK();
 }
 
-void _unpack_pivots_internal_kernel(
-  TensorIterator& iter,
-  int64_t dim_size
-) {
-  if (iter.numel() == 0) {
-    return;
-  }
-
+void unpack_pivots_cuda_kernel(TensorIterator& iter, const int64_t dim_size) {
   if (!iter.can_use_32bit_indexing()) {
     for (auto& sub_iter : iter.with_32bit_indexing()) {
-      _unpack_pivots_internal_kernel(sub_iter, dim_size);
+      unpack_pivots_cuda_kernel(sub_iter, dim_size);
     }
     return;
   }
 
-  auto offset_calculator = make_offset_calculator<2>(iter);
+  const auto offset_calculator = make_offset_calculator<2>(iter);
 
-  char* unpacked_pivots_ptr = reinterpret_cast<char*>(iter.data_ptr(0));
-  const char* const __restrict__ pivots_ptr = reinterpret_cast<const char*>(iter.data_ptr(1));
+  const auto perm_ptr = reinterpret_cast<char*>(iter.data_ptr(0));
+  const auto pivots_ptr = reinterpret_cast<const char*>(iter.data_ptr(1));
 
-  auto loop = [=]C10_DEVICE(int i) {
-    auto offsets = offset_calculator.get(i);
+  auto loop = [=]C10_DEVICE(const int idx) {
+    const auto offsets = offset_calculator.get(idx);
```

Comment:
You are obsessed with `const` :)

Reply:
That is not false :)
I mean, you may get free optimisations / warnings from the compiiler so why not?